Use structopt for cli options

hardliner66 opened the pull request:
This makes the cli handling more type safe and should simplify adding new arguments.

I also added the base support for generating shell completions. Currently it generates the completion files into a specified directory (default: ./completions). I currently haven't added any documentation for it, because i still want to do a few tests and improve the generation (e.g.: auto detect the correct directory for the chosen shell).

cafce25 commented on src/tags.rs:
```diff
@@ -1,76 +1,63 @@
+    #[structopt(short)]
+    long: bool,
+    #[structopt(short, long)]
+    search_string: Option<String>,
```
I would prefer having this as positional argument since I'd rarely use list-all without it.
But I don't think it's absolutely necessary.

hardliner66 replied:
I made it positional, but had to change the argument order a bit. I also renamed it to filter, because it's a more fitting name.

cafce25 reviewed (approved):
Totally agree on using structopt over clap. TIL about structopt.
For some reason when i searched for commandline argument parsing only clap showed up.
I'll let you decide if you want to change the search-string back to a positional argument.
Thanks!

hardliner66 commented:
I added a few help/about strings and renamed some properties. I also added options for search.
The filter (renamed from search_string) is now positional, as it fits better with the intended use case. (Thanks for the input @cafce25).

cafce25 reviewed (approved):
I like it, good work @hardliner66 !
